source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst

```diff
@@ -355,29 +397,85 @@ Before a connection can be returned outside of the pool, it must be “set up”
 Closing a Connection (Internal Implementation)
 ----------------------------------------------
 
-When a Connection is closed, it MUST first be marked as closed, removing it from being counted as available, in use, or setup. Once the connection is marked as closed, the Connection can perform whatever teardown is necessary. The Driver SHOULD perform the teardown in a non-blocking manner.
+When a `Connection <#connection>`_ is closed, it MUST first be marked as "closed",
+removing it from being counted as "available" or "in use". One that is
+complete, the `Connection <#connection>`_ can perform whatever teardown is
+necessary to close its underlying socket. The Driver MUST perform this
+teardown in a non-blocking manner, such as via the use of a background
+thread or async I/O.
 
 .. code::
 
-    mark connection as closed
+    decrement total connection count
+    if connection state is "available":
+      decrement available connection count
+    set connection state to "closed"
```

Comment:
I believe the state should be changed first, otherwise connection accounting is going to be incorrect. It is more sensible to have some tracked connections that are closed than to have disappearing open connections.